library/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -123,6 +124,16 @@ private void addDialogPickerListeners(FragmentManager fragmentManager) {
                     materialTextInputLayout.getEditText().setOnClickListener(datePickerListener);
                     materialTextInputLayout.getEditText().setOnFocusChangeListener(datePickerListener);
 
+                }
+            } else if(v instanceof MaterialTextInputLayout && v.getTag(R.id.type).equals(JsonFormConstants.TIME_PICKER)){
```

Comment:
I know that it has been required to add the listeners for the pickers later on the process and not during the widget creation as we did before. 
My worry about this is that part of the widget logic is outside of the widget factory. Is there any mechanism to put this logic back into the picker widget factory?

Reply:
Yes, that's right, we have been required o add the listeners later on the process, because we cannot use the FragmentManager before the fragment is created. I'm not sure if there is any mechanism that allows to do this in such way.